deploy-service/universal/src/main/java/com/pinterest/teletraan/universal/http/HttpClient.java

```diff
@@ -182,13 +187,37 @@ String makeCall(Request request) throws IOException {
             int responseCode = response.code();
             String responseBody = response.body() != null ? response.body().string() : "";
 
-            if (response.isSuccessful()) {
-                return responseBody;
-            } else if (responseCode >= 400 && responseCode < 500) {
-                throw new ClientErrorException(responseBody, responseCode);
-            } else {
-                throw new ServerErrorException(responseBody, responseCode);
+            if (!response.isSuccessful()) {
+                mapResponseToException(responseCode, responseBody);
             }
+            return responseBody;
+        }
+    }
+
+    protected void mapResponseToException(int responseCode, String responseBody)
+            throws WebApplicationException {
+        log.info(
+                "Egress HTTP failed with response code: {}, response body: {}",
+                responseCode,
+                responseBody);
+
+        switch (responseCode) {
+            case 400:
+                throw new BadRequestException(responseBody);
+            case 401:
+                throw new ForbiddenException(responseBody);
+            case 403:
+                throw new NotAuthorizedException(responseBody);
```

Comment:
Seems the exceptions here should be flipped
* [`ForbiddenException`](https://docs.oracle.com/javaee/7/api/javax/ws/rs/ForbiddenException.html) - 403
* [`NotAuthorizedException`](https://docs.oracle.com/javaee/7/api/javax/ws/rs/NotAuthorizedException.html) - 401 Unauthorized

Reply:
Great, fixed